### DISCUSSION TITLE
webservice

### DISCUSSION THREAD

**Discussion by claethem** [SalesforceCommerceCloud/sfcc-ci#534] [Team Posts]
Hello Guys,

I can not send a json with webservice

```
ServiceRegistry.configure("WSTEST", {

    createRequest: function(svc, requestParameters, token, req, method){
        
        svc.requestMethod = method;
    
        
        svc.addHeader("Authorization",svc.configuration.credential.user+' '+token);
        
        svc.addHeader("Accept",'application/json');

            for (key in requestParameters){
                svc.addParam(key, requestParameters.get(key));
            }
        req=JSON.stringify({'FirstName':'test','LastName':'test'});      
    	return req;

    },
	parseResponse : function(svc, res ) {
        var result = JSON.parse(res.text);
        return result;
	}
});
```

not work.

the problem is here: 
`req=JSON.stringify({'FirstName':'test','LastName':'test'});      
    	return req;`

how i can add a json in the body?

a example?

Best Regards,
Christopher

**Comment by mdzafar** [comment]
@claethem could you please post the error you are getting in log file ?

**Comment by claethem** [comment]
the error is generated by the API (Error 500), the connection is done well. But the API does not receive the json so it answers a 500 error.

**Comment by ghost** [comment]
Did you try your JSON string in a parser?
{'FirstName':'test','LastName':'test'} seems to be not valid.
Putting it in at http://json.parser.online.fr brings up an error.
Maybe you can try by using this syntax:
{"FirstName":"test","LastName":"test"}

There's a chance, that your webservice server does not understand the single quote json string.

**Comment by claethem** [comment]
Thanx i change with {"FirstName":"test","LastName":"test"}.
But i have again the error 500.  

here the "req" is the body ? there isn't a method setBody?

**Comment by Daniel-ivanov-** [comment]
@claethem if this is http service you can just do add param function.
service.addParam('json', yourjsonhere);
